src/test/java/com/wikia/webdriver/testcases/adstests/TestNoAdsForUsers.java

```diff
@@ -0,0 +1,26 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.geoedge.CountryCode;
+import com.wikia.webdriver.common.core.geoedge.GeoEdgeBrowserMobProxy;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.testng.annotations.Test;
+
+public class TestNoAdsForUsers extends TemplateNoFirstLoad {
+
+  @Test(
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "noAdsForUsers",
+      groups = "NoAdsForUsers"
+  )
+  @Execute(asUser = User.USER)
+  public void noAdsForUsers(String wikiName, String path) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00112)